core/src/main/scala/org/apache/spark/sql/delta/DeltaOptions.scala

```diff
@@ -93,6 +94,14 @@ trait DeltaWriteOptionsImpl extends DeltaOptionParser {
     options.get(DATA_CHANGE_OPTION).exists(!toBoolean(_, DATA_CHANGE_OPTION))
   }
 
+  /** Whether to only overwrite partitions that have data written into it at runtime. */
+  def isDynamicPartitionOverwriteMode: Boolean = {
+    val mode = options.get(PARTITION_OVERWRITE_MODE)
+      .getOrElse(sqlConf.getConf(SQLConf.PARTITION_OVERWRITE_MODE))
+    require(mode.equalsIgnoreCase("STATIC") || mode.equalsIgnoreCase("DYNAMIC"),
```

Comment:
i wasnt sure if validatePartitionOverwriteMode should return `Unit` (so just do the asserts) or return `Boolean` (so do the asserts and the `mode.equalsIgnoreCase("DYNAMIC")`
i pushed update that does both and returns the `Boolean`

Reply:
I was thinking `def validatePartitionOverwriteMode(): Unit`. And then instead of calling it as part of `isDynamicPartitionOverwriteMode`, just call it in `DeltaWriteOptionsImpl` so it's validated when `DeltaWriteOptionsImpl` is instantiated

Reply:
oh i see. i can do that.
how about making `isDynamicPartitionOverwriteMode` a `val` instead of a `def`? that way its also validated when `DeltaWriteOptionsImpl` is instantiated. i see this pattern in DeltaReadOptions

Reply:
Yeah that works for me